From c7dc165c2993cbd3656660c8b5484590f3667853 Mon Sep 17 00:00:00 2001
From: Juozas Kaziukenas <juozas@juokaz.com>
Date: Mon, 31 May 2010 00:49:19 +0100
Subject: [PATCH] Fixed affected rows count

---
 .../DBAL/Driver/Sqlsrv/SqlsrvStatement.php         |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lib/Doctrine/DBAL/Driver/Sqlsrv/SqlsrvStatement.php b/lib/Doctrine/DBAL/Driver/Sqlsrv/SqlsrvStatement.php
index f219982..f99dd16 100644
--- a/lib/Doctrine/DBAL/Driver/Sqlsrv/SqlsrvStatement.php
+++ b/lib/Doctrine/DBAL/Driver/Sqlsrv/SqlsrvStatement.php
@@ -244,6 +244,6 @@ class SqlsrvStatement implements \Doctrine\DBAL\Driver\Statement
      */
     public function rowCount()
     {
-        return sqlsrv_num_rows($this->_sth);
+        return sqlsrv_rows_affected($this->_sth);
     }    
 }
\ No newline at end of file
-- 
1.6.5.1.1367.gcd48

